src/mixings/logit_sb_mixing.cc

```diff
@@ -0,0 +1,161 @@
+#include "logit_sb_mixing.h"
+
+#include <google/protobuf/stubs/casts.h>
+
+#include <Eigen/Dense>
+#include <stan/math/prim.hpp>
+#include <vector>
+
+#include "mixing_prior.pb.h"
+#include "mixing_state.pb.h"
+#include "src/utils/proto_utils.h"
+#include "src/utils/rng.h"
+
+void LogitSBMixing::initialize() {
+  if (prior == nullptr) {
+    throw std::invalid_argument("Mixing prior was not provided");
+  }
+  auto priorcast = cast_prior();
+  num_clusters = priorcast->num_clusters();
```

Comment:
this parameter should better be called `num_components` (also in the prior), since the number of clusters is not `H` but the cardinality of the set {h: exists c_i = h} which in general is smaller than `H`.